loopy/check.py

```diff
@@ -668,27 +670,25 @@ def discard_dep_reqs_in_order(dep_reqs_to_vars, edges, order):
         :arg order: An instance of :class:`list` of instruction ids in which the
             *edges* graph is to be traversed.
         """
-        # predecessors: mapping from insn_id to its direct/indirect
-        # predecessors
-        predecessors = {}
-
-        for insn_id in order:
```

Comment:
`order` is no longer used in the function, so drop it? (Also be careful that while defining the order we compute a topological sort which also acts as a check that we don't have cyclic dependencies.)

Reply:
I ended up using `order` after all.